salt/metalk8s/orchestrate/deploy_node.sls

```diff
@@ -79,6 +79,9 @@ Run the highstate:
     - tgt: {{ node_name }}
     - highstate: True
     - saltenv: metalk8s-{{ version }}
+    {#- we set a timeout of 45s before checking for job liveness, otherwise
+        we can end up checking the job while minion is restarting #}
+    - timeout: 45
```

Comment:
What happens if the highstate completes in less than 45 seconds?

Reply:
The result will not be read by the master before the 45 seconds.
Maybe we could lower down the timeout, but with the risk of bringing back some flakies.

Reply:
Otherwise, as talked offline, we could move the minion configuration & the restart oustide of the highstate.

Reply:
> Otherwise, as talked offline, we could move the minion configuration & the restart oustide of the highstate.

We can "pre" reconfigure the minion before the highstate but the minion configuration should be part of the highstate anyway (but indeed if we pre-reconfigure the minion normally the minion will not restart in the highstate)

Reply:
> We can "pre" reconfigure the minion before the highstate but the minion configuration should be part of the highstate anyway (but indeed if we pre-reconfigure the minion normally the minion will not restart in the highstate).

Yes that seems like a good idea, just increase the timeout on the expected minion restart (while with highstate, we don't know _when_ the minion will restart, as it depends on ordering). That won't fix the highstate if run manually, but should be equivalent to the current approach.

> For me increasing the timeout here is good enough, pretty sure we will not win time by doing a call to pre-reconfigure salt minion instead of increase timeout here

I don't think the goal is to win time, but mostly to make sure it's not flaky. What if we need to add a bunch of states **before** the minion restart in the highstate? That timeout may become too small...

Reply:
From my understanding `timeout` is just the max time to wait before getting a minion status for a job so it should not change even if we add a bunch of states in the highstate. It may change if for what ever reason a minion take more time to restart

Reply:
Oh, that was not my understanding (based on @alexandre-allard-scality explanations, so I'm not 100% certain). To me, `timeout` is actually a "delay" to observe before polling _the_ job (here, the  `state.highstate` job), not all "sub-jobs". If for some reason the minion is still restarting 45 seconds after the request for running `highstate` was sent, the master will poll the job and hit the same issue. If we can guarantee that restarting the minion is the first thing we do in the `highstate`, then it's not necessary to split them.

Reply:
Yeah, actually it's used for both things.
The master is waiting `timeout` seconds before firing `find_job`, then waiting `timeout` seconds for the answer.